File: Assets/MiniGames/Memory/Scripts/MemoryCard.js
/*		credits: xtase studios - http://xtasestudios.com/scripts		*/
#pragma strict
public var cardnumber:int;
private var logic:Logic;
private var selected:boolean = false;

function Start () {
	logic = Camera.main.GetComponent(Logic);
}

function Update () {

}

public function SetMemorycard(t:Texture, number:int){
	renderer.materials[1].mainTexture = t;
	cardnumber = number;
}

public function Show(){
	if(!selected){
		selected = true;
		animation.Play("Flip_show");
		logic.CheckCards(this);
	}
}

public function Hide(){
	animation.Play("Flip_hide");
	selected = false;
}

public function RemoveCard(){
	animation.Play("Flip_hide");
	yield StartCoroutine("Remove");
}
function Remove(){
	yield WaitForSeconds(.5);
	Destroy(gameObject);
}

File: Assets/MiniGames/Memory/Scripts/MemoryCard.cs
/*		credits: xtase studios - http://xtasestudios.com/scripts		*/
using UnityEngine;
using System.Collections;

public class MemoryCard : MonoBehaviour {
	
	public int cardnumber;
	private Logic logic;
	private bool selected = false;
	
	// Use this for initialization
	void Start () {
		logic = Camera.main.GetComponent<Logic>();
	}
	
	// Update is called once per frame
	void Update () {
	
	}
	
	public void SetMemorycard(Texture t, int number){
		GetComponent<Renderer>().materials[1].mainTexture = t;
		cardnumber = number;
	}
	
	public void Show(){
		if(!selected){
			selected = true;
			GetComponent<Animation>().Play("Flip_show");
			logic.CheckCards(this);
		}
	}

	public void Hide(){
		GetComponent<Animation>().Play("Flip_hide");
		selected = false;
	}
	
	public void RemoveCard(){
		GetComponent<Animation>().Play("Flip_hide");
		StartCoroutine("Remove");
	}
	IEnumerator Remove(){
		yield return new WaitForSeconds(.5f);
		Destroy(gameObject);
	}
}